src/main.py

```diff
@@ -257,26 +252,50 @@ def internet_on():
 		return False
 
 
-def gettoken():
-	token = mc.get("access_token")
-	refresh = config['alexa']['refresh_token']
-	if token:
-		return token
-	elif refresh:
+class Token(object):
+
+	_token = None
+	_timestamp = None
+	_validity = 3570
+
+	def __init__(self, aconfig):
+
+		self._aconfig = aconfig
+
+		if not self._aconfig.get('refresh_token'):
+			logger.critical("AVS refresh_token not found in the configuration file. "
+					"Run the setup again to fix your installation (see project wiki for installation instructions).")
+			sys.exit(1)
+
+		self.renew()
+
+	def __str__(self):
+
+		if time.time() - self._timestamp > self._validity:
+			logger.debug("AVS token: Expired")
+			self.renew()
+
+		return self._token
+
+	def renew(self):
+
+		logger.info("AVS token: Requesting a new one")
+
 		payload = {
-			"client_id": config['alexa']['Client_ID'],
-			"client_secret": config['alexa']['Client_Secret'],
-			"refresh_token": refresh,
+			"client_id": self._aconfig['Client_ID'],
+			"client_secret": self._aconfig['Client_Secret'],
+			"refresh_token": self._aconfig['refresh_token'],
 			"grant_type": "refresh_token"
 		}
+
 		url = "https://api.amazon.com/auth/o2/token"
 		response = requests.post(url, data=payload)
 		resp = json.loads(response.text)
```

Comment:
I think that it would be good to add here check for response error. Just in case that something is wrong, it is better to deal with human readable error rather than with python stack trace saying that json cannot be parsed.